### PR TITLE
[Bug 19584] Allow target object to be deleted in frontscripts

### PR DESCRIPTION
**Pull request by peter-b**

### CODE REVIEW

File: engine/src/externalv1.cpp [hunk 1]
```diff
@@ -1208,12 +1208,10 @@ static MCExternalError MCExternalContextQuery(MCExternalContextQueryTag op, MCEx
             break;
         case kMCExternalContextQueryTarget:
         {
-            MCObjectHandle t_handle;
-            t_handle = MCtargetptr . object -> GetHandle();
-            if (!t_handle)
+            if (!MCtargetptr)
                 return kMCExternalErrorOutOfMemory;
```
Comment (runrevmark) [inline engine/src/externalv1.cpp]: This needs to be kMCExternalErrorObjectDoesNotExist (previously it assumed that targetptr always existed, and thus the only thing that could fail was creation of the weak handle - which always exists now, if the object does).

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 7e71ed8

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 7e71ed8905937d657132b14ef4636bd66498042f

**Comment by livecode-vulcan** [comment]
😞 test failure 7e71ed8905937d657132b14ef4636bd66498042f

* try-community-armv6-android-api8: [failure](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/158)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/160)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/146)
* try-community-universal-ios-iphoneos10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/90)
* try-community-universal-ios-iphonesimulator10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/90)
* try-community-universal-mac-macosx10.6: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/156)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/160)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/160)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/164)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/160)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/164)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/184)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/157)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok f84b490

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok f84b4901af52b0d9675e08cc792b5daeb2dff47e

**Comment by livecode-vulcan** [comment]
😞 test failure f84b4901af52b0d9675e08cc792b5daeb2dff47e

* try-community-armv6-android-api8: [failure](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/159)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/161)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/147)
* try-community-universal-ios-iphoneos10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/91)
* try-community-universal-ios-iphonesimulator10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/91)
* try-community-universal-mac-macosx10.6: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/157)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/161)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/161)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/165)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/161)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/165)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/185)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/158)

File: engine/src/mblhandlers.cpp [hunk 2]
```diff
@@ -3526,7 +3526,14 @@ Exec_stat MCHandlePick(void *context, MCParameter *p_parameters)
 
 	// call the Exec method to process the pick wheel
     // The function sets the result itself.
-	MCPickExecPickOptionByIndex(ctxt, kMCChunkTypeLine, t_option_lists . Ptr(), t_option_lists . Size(), t_indices . Ptr(), t_indices . Size(), t_use_hilite, t_use_picker, t_use_cancel, t_use_done, MCtargetptr . object -> getrect());
+	if (t_success && !MCtargetptr)
+	{
+		ctxt.LegacyThrow(EE_NOTARGET);
+		t_success = false;
+	}
+	if (t_success)
```
Comment (livecodepanos) [inline engine/src/mblhandlers.cpp]: Ah it needs a `{` here, sorry for missing that before :)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok d19f317

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok d19f31736ae4d148e6e66299115fb15be9935652

**Comment by livecode-vulcan** [comment]
😞 test failure d19f31736ae4d148e6e66299115fb15be9935652

* try-community-armv6-android-api8: [failure](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/162)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/164)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/150)
* try-community-universal-ios-iphoneos10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/94)
* try-community-universal-ios-iphonesimulator10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/94)
* try-community-universal-mac-macosx10.6: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/160)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/164)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/164)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/168)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/164)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/168)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/188)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/161)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok d7aeee2

**Comment by peter-b** [comment]
@livecode-vulcan review by @livecodepanos ok 5c8572e

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 5c8572efd943c6cedf8d206d3441c4806670eb9e

**Comment by livecode-vulcan** [comment]
😞 test failure 5c8572efd943c6cedf8d206d3441c4806670eb9e

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/165)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/167)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/153)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/97)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/97)
* try-community-universal-mac-macosx10.6: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/163)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/167)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/167)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/171)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/167)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/171)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/191)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/164)

File: engine/src/object.h [hunk 3]
```diff
@@ -33,6 +33,7 @@ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
 #if 0
 #include <type_traits>
 #endif
+#include <utility>
```
Comment (montegoulding) [inline engine/src/object.h:36]: Needs `#include <algorithm>` I believe for pre C++11.

**Pull request closed without merging**